File: epms/src/services/cost_centers.ts
import { api } from '@/lib/api'

export interface ApiCostCenter {
  id: string
  code: string
  name: string
  department_id: string
  is_active: boolean
  created_at: string
}

export interface CreateCostCenterBody {
  code: string
  name: string
  department_id: string
  is_active?: boolean
}

export interface UpdateCostCenterBody {
  code?: string
  name?: string
  department_id?: string
  is_active?: boolean
}

export interface CostCenterFilters {
  department_id?: string
  active_only?: boolean
}

// NOTE: The EPMS /cost-centers endpoints are GET-only — create/update/delete were
// moved to mdm-api. Do not re-add write methods here; call mdm-api instead.
export const costCenterService = {
  list: (filters?: CostCenterFilters) =>
    api.get<ApiCostCenter[]>('/cost-centers', filters),

  get: (id: string) =>
    api.get<ApiCostCenter>(`/cost-centers/${id}`),
}

File: epms-api/app/api/v1/cost_centers.py
"""Cost Center read-only endpoints.

Writes (POST/PATCH/DELETE) moved to mdm-api as of 2026-05-27 — see
`docs/superpowers/specs/2026-05-26-cost-center-dept-mdm-migration.md`. GETs
remain here so legacy EPMS callers (PR create/edit, Budget pages, Reports)
keep working during the transition; new callers should target
`mdm-api/mdm/v1/cost-centers`.
"""
import uuid

from fastapi import APIRouter, HTTPException, Query, status

from app.core.deps import SessionDep
from app.crud import cost_center as cc_crud
from app.schemas.department import CostCenterResponse

router = APIRouter(prefix="/cost-centers", tags=["cost-centers"])


@router.get("", response_model=list[CostCenterResponse])
async def list_cost_centers(
    db: SessionDep,
    department_id: uuid.UUID | None = Query(default=None),
    active_only: bool = False,
):
    """List cost centers, optionally filtered by department."""
    return await cc_crud.get_all(db, department_id=department_id, active_only=active_only)


@router.get("/{cc_id}", response_model=CostCenterResponse)
async def get_cost_center(cc_id: uuid.UUID, db: SessionDep):
    cc = await cc_crud.get_by_id(db, cc_id)
    if cc is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Cost center not found")
    return cc
